src/core/nginx.c

```diff
@@ -1305,6 +1319,49 @@ ngx_set_env(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
 }
 
 
+#if (T_NGX_MASTER_ENV)
+
+static char *
+ngx_master_set_env(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
+{
+    size_t       klen, vlen;
+    u_char      *key, *v, *p;
+    ngx_str_t   *value;
+
+    value = cf->args->elts;
+    p = ngx_strlchr(value[1].data, value[1].data + value[1].len, '=');
+    if (p == NULL) {
+        klen = value[1].len;
+        v = (u_char *) "1";
```

Comment:
could we set default path more kindly?

Reply:
Oh, Thx. `master_env` is an directive that sets the environment variable, so the default value is 1.
Of course, may be we should set a default path in the dns cache scenario.